package/MDAnalysis/core/accessors.py

```diff
@@ -63,10 +63,14 @@
    :members:
 
 """
+from __future__ import absolute_import
+from __future__ import annotations
 
 from functools import partial, update_wrapper
 
 from .. import _CONVERTERS
+import typing as t
+from .groups import AtomGroup
```

Comment:
Looks like CI is complaining about a circular import here: 

```
ImportError: cannot import name 'AtomGroup' from partially initialized module 'MDAnalysis.core.groups' (most likely due to a circular import)
```

This kind of thing, avoiding circular imports with type hinting, can indeed be a genuine problem. The team may have some suggestions, I know this one looks familiar: https://stackoverflow.com/a/39757388. For example, NumPy and SciPy both use the `TYPE_CHECKING` guard in select locations. Maybe something to discuss with the team/mentor a bit.

Reply:
I have already started to come across this in my work, eg on #3718 so possibly good to come to some kind of general agreement on this.

Reply:
Looks like it is the right solution, indeed. I was initially thinking about forward references, where you provide the type name as a string, but I am not sure it will work for imported stuff.

Reply:
I've typically seen it done as `if TYPE_CHECKING` too.